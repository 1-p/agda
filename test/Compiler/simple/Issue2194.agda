-- ASR (2016-09-20). This test is disabled with the JS and UHC
-- backends. After fixing to merge it into the `Floats` tests.

module _ where

open import Common.Prelude

printB : Bool â†’ IO Unit
printB true  = putStrLn "true"
printB false = putStrLn "false"

_/_  = primFloatDiv
_==_ = primFloatEquality

NaN : Float
NaN = 0.0 / 0.0

main : IO Unit
main =
  putStr "NaN == -NaN = " ,, printB (NaN == (primFloatNegate NaN)) ,,
  return unit
