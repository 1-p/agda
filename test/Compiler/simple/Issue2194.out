EXECUTED_PROGRAM

ret > ExitSuccess
out > NaN == -NaN = false
out >
